lib/cors.lua

```diff
@@ -124,6 +122,11 @@ end
 -- txn: The current transaction object that gives access to response properties.
 function cors_response(txn)
   local transaction_data = txn:get_priv()
+
+  if transaction_data == nil then
```

Comment:
This looks useful, but I have a question about why you are checking whether `transaction_data` is null. It looks to me like the line that sets it will always be called:

```
txn:set_priv(transaction_data)
```

Reply:
Well its for the following usecase.

A HAProxy frondend is bind to a port (80 or 443). If you only have 1 site then its easy. 1 fontend = 1 backend.
But if your HAProxy is serving multiple backend on the same port based on conditions you can have the case where you want to allow CORS for only 1 of those backends.

HAProxy config example
```
frontend frontend1
    mode http
    bind :443 ssl crt /usr/local/etc/haproxy/certs alpn h2,http/1.1

    ....

    #Setup CORS
    http-request lua.cors "*" "*", "*" if { req.hdr(Host) -i <domain A> }
    http-response lua.cors

    # Setup backends
    use_backend backend_1  if { req.hdr(Host) -i <domain A> }
    use_backend backend_B  if { req.hdr(Host) -i <domain B> }

    ...
```

In HAProxy its not possible (without using the same tricks as cors.lua) to do the if { req.hdr(Host) -i <domain A> } on the http-response.
So in the above case transaction_data would be nil.

Reply:
Okay, that makes sense.